src/overlays/actors/ovl_En_Ssh/z_en_ssh.c

```diff
@@ -21,95 +63,798 @@ const ActorInit En_Ssh_InitVars = {
+    } else {
+        EnSsh_Turn(this, globalCtx);
+    }
+}
+
+void EnSsh_Bob(EnSsh* this, GlobalContext* globalCtx) {
+    f32 bobVel = 0.5f;
+
+    if ((globalCtx->state.frames & 8) != 0) {
+        bobVel *= -1.0f;
+    }
+    Math_SmoothScaleMaxMinF(&this->actor.velocity.y, bobVel, 0.4f, 1000.0f, 0.0f);
+}
+
+s32 EnSsh_IsCloseToLink(EnSsh* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    f32 yDist;
+
+    if (this->stateFlags & STATE_GROUND_START) {
+        return true;
+    }
+    if (this->unkTimer != 0) {
+        return true;
+    }
+    if (this->swayTimer != 0) {
+        return true;
+    }
+    if (this->animTimer != 0) {
+        return true;
+    }
+
+    if (this->actor.xzDistFromLink > 160.0f) {
+        // player is more than 160 xz units from the skulltulla
+        return false;
+    }
+
+    yDist = this->actor.posRot.pos.y - player->actor.posRot.pos.y;
+    if (yDist < 0.0f || yDist > 400.0f) {
+        // player is above the skulltulla or more than 400 units below
+        // the skulltulla
+        return false;
+    }
+
+    if (player->actor.posRot.pos.y < this->actor.groundY) {
+        // player is below the Skulltulla's ground position
+        return false;
+    }
+    return true;
+}
+
+s32 EnSsh_IsCloseToHome(EnSsh* this) {
+    f32 vel = this->actor.velocity.y;
+    f32 nextY = this->actor.posRot.pos.y + 2.0f * this->actor.velocity.y;
+
+    if (nextY >= this->actor.initPosRot.pos.y) {
+        return 1;
+    }
+    return 0;
+}
+
+s32 EnSsh_IsCloseToGround(EnSsh* this) {
+    f32 vel = this->actor.velocity.y;
+    f32 nextY = this->actor.posRot.pos.y + 2.0f * this->actor.velocity.y;
+
+    if ((nextY - this->actor.groundY) <= this->groundYoffset) {
+        return 1;
+    }
+    return 0;
+}
+
+void EnSsh_Sway(EnSsh* this) {
+    Vec3f swayVecBase;
+    Vec3f swayVec;
+    f32 temp;
+    s16 swayAngle;
+
+    if (this->swayTimer != 0) {
+        this->swayAngle += 0x640;
+        this->swayTimer--;
+        if (this->swayTimer == 0) {
+            this->swayAngle = 0;
+        }
+        temp = this->swayTimer * (1.0f / 6);
+        swayAngle = temp * (0x10000 / 360.0f) * Math_Sins(this->swayAngle);
+        temp = this->actor.posRot.pos.y - this->ceilingPos.y;
+        swayVecBase.x = Math_Sins(swayAngle) * temp;
+        swayVecBase.y = Math_Coss(swayAngle) * temp;
+        swayVecBase.z = 0.0f;
+        Matrix_Push();
+        Matrix_Translate(this->ceilingPos.x, this->ceilingPos.y, this->ceilingPos.z, MTXMODE_NEW);
+        Matrix_RotateY(this->actor.posRot.rot.y * (M_PI / 0x8000), MTXMODE_APPLY);
+        Matrix_MultVec3f(&swayVecBase, &swayVec);
+        Matrix_Pull();
+        this->actor.shape.rot.z = -(swayAngle * 2);
+        this->actor.posRot.pos.x = swayVec.x;
+        this->actor.posRot.pos.z = swayVec.z;
+    }
+}
+
+void EnSsh_CheckBodyStickHit(EnSsh* this, GlobalContext* globalCtx) {
+    ColliderBody* body = &this->colCylinder[0].body;
```

Comment:
Is this temp necessary? The other cylinders are accessed directly so I feel like this might be an optimization (a branch head merge). The player temp might also need to be inlined in order for it to work tho, but that's fine.

Reply:
Changing it doesn't work.